cpp/src/spatial/linestring_distance.cu

```diff
@@ -38,6 +38,48 @@
 namespace cuspatial {
 namespace {
 
+template <typename T>
+vec_2d<T> __device__ operator+(vec_2d<T> const& a, vec_2d<T> const& b)
```

Comment:
Let's move vec_2d (and its derived types) to a vec_2d.hpp and put these operators in that header. Also make these `__host__ __device__` (and use a macro for that so it's portable).

Reply:
Also need to document these functions.